kubernetes/helm_charts/secor/templates/secor-statefulset.yaml

```diff
@@ -0,0 +1,124 @@
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.Name }}
+  namespace: {{ .Values.secor_namespace }}
+  labels:
+    app: secor
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  serviceName: secor
+  updateStrategy:
+    type: RollingUpdate
+  replicas: {{ dig $.Release.Name "replicas" "" $.Values.secor_jobs }}
```

Comment:
What does `dig` do? Is it compatible with helm 3? What ius the advantage of using this instead of something like `{{ .Values.replicaCount }}`

Reply:
dig function is compatible with helm 3. It is used when you have to read values from a nested dictionary. In this case, the replicas will be under secor_jobs.<job_name>.replicas and hence we cannot directly read it using .Values.replicaCount. The replicas will be configurable for each job.